ledger_device_sdk/src/nbgl.rs

```diff
@@ -492,26 +482,37 @@ impl<'a> NbglAddressReview<'a> {
     }
 
     pub fn verify_str(self, verify_str: &'a str) -> NbglAddressReview<'a> {
-        NbglAddressReview { verify_str, ..self }
+        NbglAddressReview {
+            verify_str: CStr::from_bytes_until_nul(verify_str.as_bytes())
+                .unwrap()
+                .as_ptr(),
+            ..self
+        }
     }
 
     pub fn show(&mut self, address: &str) -> bool {
         unsafe {
             // Create CStringHelper instance and convert the address and verify_str into c strings.
-            let c_string_helper = CStringHelper::<128>::new();
-            let c_addr_str = c_string_helper.to_cstring(address).unwrap();
-            let c_verif_str = c_string_helper.to_cstring(self.verify_str).unwrap();
-
-            if self.glyph.is_some() {
-                self.icon = &self.glyph.unwrap().into() as *const nbgl_icon_details_t;
-            }
+            //let c_string_helper = CStringHelper::<128>::new();
+            //let c_addr_str = c_string_helper.to_cstring(address).unwrap();
+            //let c_verif_str = c_string_helper.to_cstring(self.verify_str).unwrap();
+
+            let addr = CStr::from_bytes_until_nul(address.as_bytes())
+                .unwrap()
+                .as_ptr();
+
+            let icon = if self.glyph.is_some() {
```

Comment:
I think I tried this syntax (declaring a local variable, which is more logical in this case) but had display issues or crashes.